tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -463,58 +464,75 @@ else if (col.getColumnInfo().isPrimaryKey() && tableInfo.isPkHandle()) {
 
     if (!getGroupByItems().isEmpty() || !getAggregates().isEmpty()) {
       if (!isIndexDoubleScan || (isGroupByCoveredByIndex() && isAggregateCoveredByIndex())) {
-        Aggregation.Builder aggregationBuilder = Aggregation.newBuilder();
-        getGroupByItems()
-            .forEach(
-                tiByItem ->
-                    aggregationBuilder.addGroupBy(
-                        ProtoConverter.toProto(tiByItem.getExpr(), colOffsetInFieldMap)));
-        getAggregates()
-            .forEach(
-                tiExpr ->
-                    aggregationBuilder.addAggFunc(
-                        ProtoConverter.toProto(tiExpr, colOffsetInFieldMap)));
-        executorBuilder.setTp(ExecType.TypeAggregation);
-        dagRequestBuilder.addExecutors(executorBuilder.setAggregation(aggregationBuilder));
-        executorBuilder.clear();
-        addPushDownGroupBys();
-        addPushDownAggregates();
+        pushDownAggAndGroupBy(dagRequestBuilder, executorBuilder, colOffsetInFieldMap);
```

Comment:
How about we add some comments for better understanding?

Reply:
indded